### ISSUE TITLE
Generated Operation IDs contain unwanted characters

### ISSUE THREAD

**Issue by pvgoran** [Ajaxy/tinyspec#13]
The use of `-` as a placeholder character that replaces non-word characters of the path to produce an Operation ID seems to be a poor choice. The specification says:

```
Tools and libraries MAY use the operationId to uniquely identify an operation, therefore, it is recommended to follow common programming naming conventions.
```

And use of `-` definitely does not follow common programming naming conventions. As a result, the Swagger's code generator that I use (`swagger-codegen-cli-2.3.1.jar`) can't produce a working code for such operationIds.

Thus, I suggest changing the placeholder character to `_`.

**Comment by Ajaxy** [comment]
I agree. Would be great if you could do a PR. I would use [`_.snakeCase`](https://lodash.com/docs/4.17.11#snakeCase) here.

However, we need to avoid collisions (for example for `GET /hello_world` vs. `GET /hello/world`). Now a double dash separator (`--`) is used for it, but it makes the operation ID quite ugly. Another possible solution would be to keep track of occurrences of each operation ID and if it's `> 0`, add some postfix to a next one, like `get_hello_world` and `get_hello_world_2`.

Also, we need to use path with the param types stripped out (after merge of #12).

**Comment by pvgoran** [comment]
I agree, it would be probably better to remove the auto-generated Operation IDs altogether. The best code is no code at all!
